v2/kafka-to-bigquery/src/main/java/com/google/cloud/teleport/v2/templates/KafkaToBigQueryFlex.java

```diff
@@ -248,27 +252,74 @@ public static PipelineResult runAvroPipeline(
 
     WriteResult writeResult = null;
 
-    if (options.getAvroFormat().equals("NON_WIRE_FORMAT") && options.getAvroSchemaPath() != null) {
-
-      writeResult = kafkaRecords.apply(AvroTransform.of(options));
+    if (options.getMessageFormat().equals("AVRO_BINARY_ENCODING")
+        && options.getBinaryAvroSchemaPath() != null) {
+      writeResult =
+          kafkaRecords.apply(
+              AvroTransform.of(
+                  options.getReadBootstrapServerAndTopic().split(":")[2],
+                  options.getMessageFormat(),
+                  options.getBinaryAvroSchemaPath(),
+                  options.getConfluentAvroSchemaPath(),
+                  options.getOutputTableSpec(),
+                  options.getWriteDisposition(),
+                  options.getCreateDisposition(),
+                  options.getNumStorageWriteApiStreams(),
+                  options.getStorageWriteApiTriggeringFrequencySec(),
+                  options.getPersistKafkaKey(),
+                  options.getUseAutoSharding()));
 
     } else {
 
-      if (options.getAvroSchemaPath() != null && options.getOutputTableSpec() == null) {
-        throw new IllegalArgumentException("An output BigQuery table is required.");
-      }
-
-      if (options.getAvroSchemaPath() != null && options.getOutputTableSpec() != null) {
-        writeResult = kafkaRecords.apply(AvroTransform.of(options));
-      }
-
-      if (options.getSchemaRegistryConnectionUrl() != null && options.getOutputDataset() == null) {
+      if (options.getSchemaFormat().equals("SINGLE_SCHEMA_FILE")) {
+
+        if (options.getConfluentAvroSchemaPath() != null && options.getOutputTableSpec() == null) {
+          throw new IllegalArgumentException("An output BigQuery table is required.");
```

Comment:
I'd add more details to the error message.
"outputTableSpec parameter is required when using SINGLE_SCHEMA_FILE schema format and <... what are the other conditions?...>"

Reply:
done